NitroxClient/MonoBehaviours/MultiplayerButton.cs

```diff
@@ -0,0 +1,234 @@
+            multiplayerButtonInst.transform.Find("NewGameButton/Text").GetComponent<Text>().text = text;
+            Button multiplayerButtonButton = multiplayerButtonInst.transform.Find("NewGameButton").GetComponent<Button>();
+            multiplayerButtonButton.onClick = new Button.ButtonClickedEvent();
+            multiplayerButtonButton.onClick.AddListener(clickEvent);
+            multiplayerButtonInst.transform.SetParent(savedGameAreaContent, false);
+
+            GameObject delete = Instantiate(savedGamesRef.GetComponent<MainMenuLoadPanel>().saveInstance.GetComponent<MainMenuLoadButton>().deleteButton);
+            Button deleteButtonButton = delete.GetComponent<Button>();
+            deleteButtonButton.onClick = new Button.ButtonClickedEvent();
+            deleteButtonButton.onClick.AddListener(deleteEvent);
+            delete.transform.SetParent(multiplayerButtonInst.transform, false);
+        }
+
+        public void JoinServer(string serverIp)
+        {
+            new GameObject().AddComponent<JoinServerScript>().serverIp = serverIp;
+        }
+
+        public void CreateNewServer()
+        {
+            showingAddServer = true;
+        }
+
+        public void AddServer()
+        {
+            using (StreamWriter sw = new StreamWriter(SERVER_LIST_PATH, true))
+            {
+                sw.WriteLine($"{serverInput}|{ipInput}");
+            }
+        }
+
+        public void RemoveServer(int index)
+        {
+            List<string> serverLines = new List<string>(File.ReadAllLines(SERVER_LIST_PATH));
+            serverLines.RemoveAt(index - 1);
+            File.WriteAllLines(SERVER_LIST_PATH, serverLines.ToArray());
+            Destroy(savedGameAreaContent.GetChild(index).gameObject);
+        }
+
+        public void HideNewServer()
+        {
+            serverInput = "server name";
+            ipInput = "server ip";
+            showingAddServer = false;
+        }
+
+        public void OnGUI()
+        {
+            if (!showingAddServer)
+            {
+                return;
+            }
+            serverInput = GUI.TextField(new Rect(Screen.width / 2 - 250, Screen.height / 2 - 50, 500, 50), serverInput);
+            ipInput = GUI.TextField(new Rect(Screen.width / 2 - 250, Screen.height / 2, 500, 50), ipInput);
+            if (GUI.Button(new Rect(Screen.width / 2 - 250, Screen.height / 2 + 50, 500, 50), "Add server"))
+            {
+                AddServer();
+                int serverIndex = savedGameAreaContent.childCount;
+                CreateServerButton($"<b>{serverInput}</b>\n{ipInput}", delegate { JoinServer(ipInput); }, delegate { RemoveServer(serverIndex); });
+                HideNewServer();
+            }
+            if (GUI.Button(new Rect(Screen.width / 2 - 250, Screen.height / 2 + 100, 500, 50), "Cancel"))
+            {
+                HideNewServer();
+            }
+        }
+    }
+
+    public class JoinServerScript : MonoBehaviour
```

Comment:
minor: would recommend dropping the word Script as it is implied by the monobehaviour namespace.  Maybe brainstorm a more descriptive postfix(Gui maybe?)